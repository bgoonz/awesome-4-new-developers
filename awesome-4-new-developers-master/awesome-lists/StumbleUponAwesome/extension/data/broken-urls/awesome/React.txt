http://blog.lum.pe/gesture-detection-in-react-native-fixing-unexpected-panning/,Gesture detection in React Native â€“ fixing unexpected panning,https://github.com/enaqx/awesome-react#readme,React error: URLError(gaierror(8, 'nodename nor servname provided, or not known')) | reason: [Errno 8] nodename nor servname provided, or not known
http://atticuswhite.com/blog/react-native-smart-mirror-lab/,Digital Smart Mirror lab with React Native,https://github.com/enaqx/awesome-react#readme,React error: timeout('The read operation timed out') | reason: 