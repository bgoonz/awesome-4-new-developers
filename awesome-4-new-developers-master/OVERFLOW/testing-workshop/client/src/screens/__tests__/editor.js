import React from 'react'
import {
  generate,
  wait,
  cleanup,
  fireEvent,
  renderIntoDocument,
  render,
} from 'til-client-test-utils'
import Editor from '../editor'

afterEach(cleanup)

test('calls onSubmit with the username and password when submitted', async () => {
  // Arrange
  const fakeUser = generate.userData({id: generate.id()})
  const fakePost = generate.postData({authorId: fakeUser.id})
  const fakeHistory = {push: jest.fn()}
  const fakeApi = {
    posts: {
      create: jest.fn(() => Promise.resolve()),
    },
  }
  const {getByText, getByLabelText} = renderIntoDocument(
    <Editor api={fakeApi} user={fakeUser} history={fakeHistory} />,
  )

  getByLabelText('Title').value = fakePost.title
  getByLabelText('Content').value = fakePost.content
  getByLabelText('Tags').value = fakePost.tags.join(', ')
  const preDate = Date.now()

  // Act
  fireEvent.click(getByText('submit'))

  // Assert
  expect(fakeApi.posts.create).toHaveBeenCalledTimes(1)
  expect(fakeApi.posts.create).toHaveBeenCalledWith({
    ...fakePost,
    date: expect.any(String),
  })

  const postDate = Date.now()
  await wait(() => expect(fakeHistory.push).toHaveBeenCalledTimes(1))
  expect(fakeHistory.push).toHaveBeenCalledWith('/')
  const date = new Date(fakeApi.posts.create.mock.calls[0][0].date).getTime()
  expect(date).toBeGreaterThanOrEqual(preDate)
  expect(date).toBeLessThanOrEqual(postDate)
})

test('snapshot', () => {
  const {container} = render(<Editor />)
  expect(container.firstChild).toMatchSnapshot()
})
