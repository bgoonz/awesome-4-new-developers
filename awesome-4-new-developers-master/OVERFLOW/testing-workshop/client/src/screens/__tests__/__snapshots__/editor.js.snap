// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.css-0,
[data-css-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-1,
[data-css-1] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  min-width: 200px;
  max-width: 400px;
  margin-left: -116px;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-2,
[data-css-2] {
  display: grid;
  grid-template-columns: 100px 1fr;
  grid-gap: 16px;
  font-size: 18px;
  align-items: center;
  margin-top: 30px;
  margin-bottom: 30px;
  width: 100%;
  -ms-grid-template-columns: 100px 1fr;
  -ms-grid-gap: 16px;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-3,
[data-css-3] {
  background: white;
  height: 50px;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  border-bottom: 5px;
  width: 100%;
  min-width: 150px;
  display: block;
  padding-left: 10px;
}

.css-3::-webkit-input-placeholder,
[data-css-3]::-webkit-input-placeholder {
  opacity: 0.5;
}

.css-3::-moz-placeholder,
[data-css-3]::-moz-placeholder {
  opacity: 0.5;
}

.css-3::-ms-input-placeholder,
[data-css-3]::-ms-input-placeholder {
  opacity: 0.5;
}

.css-3::placeholder,
[data-css-3]::placeholder {
  opacity: 0.5;
}

.css-8,
[data-css-8] {
  background: white;
  border: none;
  height: 200px;
  border-radius: 10px;
  box-shadow: var(--shadow);
  border-bottom: 5px;
  width: 100%;
  min-width: 150px;
  display: block;
  margin: 0 auto 10px auto;
  padding-top: 10px;
  padding-left: 10px;
}

.css-8::-webkit-input-placeholder,
[data-css-8]::-webkit-input-placeholder {
  opacity: 0.5;
}

.css-8::-moz-placeholder,
[data-css-8]::-moz-placeholder {
  opacity: 0.5;
}

.css-8::-ms-input-placeholder,
[data-css-8]::-ms-input-placeholder {
  opacity: 0.5;
}

.css-8::placeholder,
[data-css-8]::placeholder {
  opacity: 0.5;
}

.css-13,
[data-css-13] {
  font-size: 13px;
  font-family: Raleway,sans-serif;
  background: var(--green);
  padding: 10px 20px;
  display: block;
  margin-left: auto;
  color: white;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  cursor: pointer;
  transition: 0.5s;
  -webkit-transition: 0.5s;
  -moz-transition: 0.5s;
}

.css-13:hover,
[data-css-13]:hover {
  box-shadow: var(--shadowHover);
}

@media only screen and (max-width: 819px) {
  .css-1,
  [data-css-1] {
    margin-left: 0;
    width: 90%;
  }
}

<div>
  <div
    class="css-0"
  >
    <form
      class="css-1"
    >
      <div
        class="css-2"
      >
        <label
          for="title-input"
          style="justify-self: right;"
        >
          Title
        </label>
        <input
          class="css-3"
          id="title-input"
          name="title"
          placeholder="Title"
        />
        <label
          for="content-input"
          style="justify-self: right; align-self: baseline;"
        >
          Content
        </label>
        <textarea
          class="css-8"
          id="content-input"
          name="content"
          placeholder="Content"
        />
        <label
          for="tags-input"
          style="justify-self: right;"
        >
          Tags
        </label>
        <input
          class="css-3"
          id="tags-input"
          name="tags"
          placeholder="tags"
        />
      </div>
      <button
        class="css-13"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
